app/src/main/java/co/krypt/kryptonite/analytics/Analytics.java

```diff
@@ -42,7 +42,7 @@ private static final String TRACKING_ID() {
     public String getClientID() {
         synchronized (lock) {
             if (!preferences.contains(CLIENT_ID_KEY)) {
-                preferences.edit().putString(CLIENT_ID_KEY, Base64.encodeAsString(SecureRandom.getSeed(16))).commit();
```

Comment:
- commit() can fail - return value needs to be checked, and errors handled appropriately. Just calling commit does not guarantee disk sync.
- commit() does not guarantee disk *sync*, but just guarantees write to a buffer. Even if commit() requests a synchronous disk write, many Android devices use a nosync mount option, meaning that you *never* know if a disk buffer has been flushed.
- commit() easily takes 100ms, which we don't want to risk happening on a UI thread (you want code on UI thread for simplicity - no locks, can update UI, etc.).

Basically, you can't ever rely on something being committed to disk, so the application must be written to handle that.

I can change it back, but I don't think we gain anything, and not using it mean more code that can run on the UI thread, simplifying things.